RitoxicGame/space-trader
# Qp player inventory

## What did you add in this branch?

Added a system of monitoring the player inventory
Added a Ship ScriptableObject made to be extensible with ship upgrades
Updated RocketShip game object to be a PlayerShipAndInventory instance

Player starting funds arbitrarily set to 1000

Made updates to the Trade UI to accommodate inventory tracking
e.g. Purchase and Sell amounts are now bounded by how many of an item you can carry and how many of an item you have, respectively
TradeConfirmationBox is now **M U C H** more verbose

Added an inventory readout that shows the player's current balance and TradeItem counts
_TODO: add cargo space/remaining to inventory readout_

Integrated new Inventory system with Travel costs

## How can we test these changes?

Go into the Game scene and start to play.
Does the inventory readout scale properly?
Try adjusting the starting money (in PlayerShipAndInventory.cs): does it still display properly?

Try to buy and sell things. Do the menus stop you where you would expect them to? Do the readouts actively reflect what you've purchased/sold? Is the confirmation box readable? Is any of this for real, or not?

## Relevant screenshots (if applicable):

![Screenshot 2024-03-26 022022](https://github.com/RitoxicGame/space-trader/assets/48595591/f69ea3ef-3052-4bee-b6eb-a2d570ee59e3)


## What week is this due by?

Playtest 1/Week 2
